contracts/src/Gateway.sol

```diff
@@ -133,45 +133,48 @@ contract Gateway is IGateway, IInitializable {
         // Otherwise malicious relayers can break the bridge by allowing the message handlers below to run out gas and fail silently.
         // In this scenario case, the channel's state would have been updated to accept the message (by virtue of the nonce increment), yet the actual message
         // dispatch would have failed
-        if (gasleft() < DISPATCH_GAS + BUFFER_GAS) {
+
+        uint256 computedGas = computeGas(message.command, message.params);
+
+        if (gasleft() < computedGas + BUFFER_GAS) {
             revert NotEnoughGas();
         }
```

Comment:
Since `dynamicGas` is provided by the end-user manipulated with an unrealistic high value, then the check here will always fail, and the bridge stuck.

Reply:
That's a good point, thanks @yrong!

Reply:
@yrong although, `dynamicGas` is only checked for AgentExecute.Transact: https://github.com/Snowfork/snowbridge/pull/924/files#diff-74470ba8f35b42dbc8805739e92708c0696343e0b1819feb3e7a5ebb77cc623cR471. So while the user can manipulate it with an unrealistic high value, it will only work for `Transact` requests, so I think this will not be a problem. Let me know if you agree.

Reply:
AFAIK currently we require each message to be handled properly without error or subsequent message will fail in `nonce` check.

So I think maybe require some extra check in `bridgehub` when forwarding the transact command, make sure not over the upper limit here.
https://github.com/Snowfork/snowbridge/blob/1201e1db5aa6ae580123a745bc7387c2019c4321/web/packages/test/scripts/set-env.sh#L27C1-L27C14